File: classicemu/serverlogonproof.py
class ServerLogonProof:
    cmd = 1
    error = 0
    M2 = None

    def __init__(self, srp):
        self.srp = srp

    def get(self, M1):
        if self.srp.getM(M1):
            data = [1, 0]
            for m in self.srp.M:
                data.append(m)
            data.append(0x00)
            data.append(0x00)
            data.append(0x00)
            data.append(0x00)
        else:
            data = [0, 0, 4]
        return data

File: src/Auth/Challenges/ServerLogonProof.cs
using Classic.Auth.Cryptography;
using Classic.Auth.Data.Enums;
using Classic.Shared;
using Classic.Shared.Data;

namespace Classic.Auth.Challenges
{
    public class ServerLogonProof
    {
        private readonly SecureRemotePasswordProtocol srp;
        private readonly int build;

        public ServerLogonProof(SecureRemotePasswordProtocol srp, int build)
        {
            this.srp = srp;
            this.build = build;
        }

        public byte[] Get(byte[] clientPublicValue, byte[] clientProof)
        {
            if (this.srp.ValidateClientProof(clientPublicValue, clientProof))
            {
                using var packet = new PacketWriter();
                packet
                    .WriteUInt8( /* cmd   */ (byte)Opcode.LoginProof)
                    .WriteUInt8( /* error */ (byte)AuthenticationStatus.Success)
                    .WriteBytes( /* M[20] */ this.srp.M)
                    .WriteUInt32(/* unk   */ 0);

                if (this.build > ClientBuild.Vanilla)
                {
                    packet
                        .WriteUInt32(/* unk2 */ 0)
                        .WriteUInt16(/* unk3 */ 0);
                }

                return packet.Build();
            }

            // Failed
            return new byte[] { 0, 0, 4 };
        }
    }
}